akka-stream/src/main/scala/akka/stream/scaladsl/Source.scala

```diff
@@ -262,6 +262,20 @@ object Source {
     fromGraph(new FutureSource(future.toScala))
 
   /**
+   * A graph with the shape of a source logically is a source.
+   * This method makes an asynchronous graph of such shape in type with
+   * an asynchronous materialized value.
+   */
+  def fromFutureGraph[T, M](future: Future[Graph[SourceShape[T], M]]): Source[T, Future[M]] = fromGraph(new FutureFlattenSource(future))
```

Comment:
Instead of a `Future[Graph]` these APIs must expect a factory: `() => Future[Graph[]]`, otherwise these are not rematerializable.